tarantool/dbapi.py

```diff
@@ -0,0 +1,251 @@
+        else:
+            self._lastrowid = -1
+
+    def executemany(self, query, param_sets):
+        self._check_not_closed("Can not execute on closed cursor.")
+        rowcount = 0
+        for params in param_sets:
+            self.execute(query, params)
+            if self.rowcount == -1:
+                rowcount = -1
+            if rowcount != -1:
+                rowcount += self.rowcount
+        self._rowcount = rowcount
+
+    @property
+    def lastrowid(self):
+        """
+        This read-only attribute provides the rowid of the last modified row
+        (most databases return a rowid only when a single INSERT operation is
+        performed).
+        """
+        return self._lastrowid
+
+    @property
+    def rowcount(self):
+        """
+        This read-only attribute specifies the number of rows that the last
+        .execute*() produced (for DQL statements like SELECT) or affected (
+        for DML statements like UPDATE or INSERT).
+        """
+        return self._rowcount
+
+    def _check_result_set(self, error=None):
+        """
+        Closed method for raising an error when Cursor object does not have
+        any row to fetch. Useful for checking access after DQL requests.
+        """
+        if self._rows is None:
+            raise InterfaceError(error or "No result set to fetch from")
+
+    def fetchone(self):
+        """
+        Fetch the next row of a query result set, returning a single
+        sequence, or None when no more data is available.
+        """
+        self._check_result_set()
+        return self.fetchmany(1)[0] if self._rows else None
+
+    def fetchmany(self, size=None):
+        """
+        Fetch the next set of rows of a query result, returning a sequence of
+        sequences (e.g. a list of tuples). An empty sequence is returned when
+        no more rows are available.
+        """
+        self._check_result_set()
+
+        size = size or self.arraysize
+
+        if len(self._rows) < size:
+            items = self._rows
+            self._rows = []
+        else:
+            items, self._rows = self._rows[:size], self._rows[size:]
+
+        return items
+
+    def fetchall(self):
+        """Fetch all (remaining) rows of a query result, returning them as a
+        sequence of sequences (e.g. a list of tuples). Note that the cursor's
+        arraysize attribute can affect the performance of this operation.
+        """
+        self._check_result_set()
+
+        items = self._rows
+        self._rows = []
+        return items
+
+    def setinputsizes(self, sizes):
+        """PEP-249 allows to not implement this method and do nothing."""
+
+    def setoutputsize(self, size, column=None):
+        """PEP-249 allows to not implement this method and do nothing."""
+
+
+class Connection(BaseConnection):
+
+    def __init__(self, *args, **kwargs):
+        super(Connection, self).__init__(*args, **kwargs)
+        self._autocommit = kwargs.get('autocommit')
```

Comment:
So the default is `None`? Hmm. But it is stated below that the default is `True`.

Reply:
I completely lost __init__ error handling for the autocommit value. To solve this I implemented the non-public `_set_autocommit(autocommit)` method which does `autocommit` value error handling. It is executed in the `autocommit` setter and in the Connection `__init__`